app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt

```diff
@@ -164,5 +222,14 @@ object ThreadController {
             }
         }
     }
+
+    fun saveThreads(threads: List<Thread>, searchMessages: List<Message>) {
+        defaultRealm.writeBlocking {
+            with(FolderController.getOrCreateSearchFolder(this)) {
+                this.messages = searchMessages.toRealmList()
+                this.threads = threads.toRealmList()
+            }
+        }
+    }
```

Comment:
```suggestion
    fun saveThreads(searchThreads: List<Thread>, searchMessages: List<Message>) {
        defaultRealm.writeBlocking {
            FolderController.getOrCreateSearchFolder(this).apply {
                messages = searchMessages.toRealmList()
                threads = searchThreads.toRealmList()
            }
        }
    }
```

Reply:
Maybe we could even just pass searchMessages, as the threads are given by `searchMessages.convertToSearchThreads()`